pkg/controller/utils/elasticsearch.go

```diff
@@ -62,3 +112,266 @@ func GetElasticsearchClusterConfig(ctx context.Context, cli client.Client) (*ren
+			"phases": map[string]interface{}{
+				"hot": map[string]interface{}{
+					"actions": map[string]interface{}{
+						"rollover": map[string]interface{}{
+							"max_size": pd.rolloverSize,
+							"max_age":  pd.rolloverAge,
+						},
+						"set_priority": map[string]interface{}{
+							"priority": 100,
+						},
+					},
+				},
+				"warm": map[string]interface{}{
+					"actions": map[string]interface{}{
+						"readonly": map[string]interface{}{},
+						"set_priority": map[string]interface{}{
+							"priority": 50,
+						},
+					},
+				},
+				"delete": map[string]interface{}{
+					"min_age": pd.deleteAge,
+					"actions": map[string]interface{}{
+						"delete": map[string]interface{}{},
+					},
+				},
+			},
+		},
+	}
+	return pd
+}
+
+func applyILMPolicy(ctx context.Context, esClient *elastic.Client, indexName string, policy map[string]interface{}) error {
+	policyName := indexName + "_policy"
+	_, err := esClient.XPackIlmPutLifecycle().Policy(policyName).BodyJson(policy).Do(ctx)
+	if err != nil {
+		log.Error(err, "Error applying Ilm Policy")
+		return err
+	}
+	return nil
+}
+
+// calculateRolloverSize returns max_size to rollover
+// max_size is based on the disk space allocated for the log type divided by ElasticsearchRetentionFactor
+// If calculated max_size is greater than ES recommended shard size (DefaultMaxIndexSizeGi), set it to DefaultMaxIndexSizeGi
+func calculateRolloverSize(totalEsStorage int64, diskPercentage float64, diskForLogType float64) string {
+	rolloverSize := int64((float64(totalEsStorage) * diskPercentage * diskForLogType) / ElasticsearchRetentionFactor)
+	rolloverMax := resource.MustParse(fmt.Sprintf("%dGi", DefaultMaxIndexSizeGi))
+	maxRolloverSize := rolloverMax.Value()
+
+	if rolloverSize > maxRolloverSize {
+		rolloverSize = maxRolloverSize
+	}
+
+	return fmt.Sprintf("%db", rolloverSize)
+}
+
+// calculateRolloverAge returns max_age to rollover
+// max_age to rollover an index is retention period set in LogStorage divided by ElasticsearchRetentionFactor
+// If retention is < ElasticsearchRetentionFactor, set rollover age to 1 day
+// if retention is 0 days, rollover every 1 hr - we dont want to rollover index every few ms/s set it to 1hr similar to curator cronjob interval
+func calculateRolloverAge(retention int) string {
+	var age string
+	if retention <= 0 {
+		age = "1h"
+	} else if retention < ElasticsearchRetentionFactor {
+		age = "1d"
+	} else {
+		rolloverAge := retention / ElasticsearchRetentionFactor
+		age = fmt.Sprintf("%dd", rolloverAge)
+	}
+	return age
+}
+
+func getClientCredentials(client client.Client, ctx context.Context) (string, string, *x509.CertPool, error) {
+	esSecret := &corev1.Secret{}
+	if err := client.Get(ctx, types.NamespacedName{Name: render.ElasticsearchOperatorUserSecret, Namespace: render.OperatorNamespace()}, esSecret); err != nil {
+		if !errors.IsNotFound(err) {
+			return "", "", nil, err
+		}
+		log.Info("Elasticsearch public cert secret not found yet")
```

Comment:
Why do we continue here? Don't we have to have this secret?
Also isn't this log message incorrect? It should be `tigera-operator/<ElasticsearchOperatorUserSecret>`